applications/GeoMechanicsApplication/custom_conditions/U_Pw_normal_lysmer_absorbing_condition.cpp

```diff
@@ -25,190 +25,218 @@ Condition::Pointer UPwLysmerAbsorbingCondition<TDim,TNumNodes>::Create(IndexType
+{
+    BoundedMatrix<double, N_DOF, N_DOF> stiffness_matrix;
 
-    // no righthand side contribution
-    if (rRightHandSideVector.size() != CONDITION_SIZE)
-        rRightHandSideVector.resize(CONDITION_SIZE, false);
-    noalias(rRightHandSideVector) = ZeroVector(CONDITION_SIZE);
+    this->CalculateConditionStiffnessMatrix(stiffness_matrix, rCurrentProcessInfo);
+
+    MatrixType global_stiffness_matrix = ZeroMatrix(CONDITION_SIZE, CONDITION_SIZE);
+    GeoElementUtilities::AssembleUBlockMatrix< TDim, TNumNodes >(global_stiffness_matrix, stiffness_matrix);
+
+    this->CalculateAndAddRHS(rRightHandSideVector, global_stiffness_matrix);
 }
 
-/// <summary>
-/// Calculates LHS Damping part of abosrbing boundary
-/// </summary>
-/// <param name="rDampingMatrix"></param>
-/// <param name="CurrentProcessInfo"></param>
+
 template< unsigned int TDim, unsigned int TNumNodes >
-void UPwLysmerAbsorbingCondition<TDim, TNumNodes>::CalculateDampingMatrix(MatrixType& rDampingMatrix, const ProcessInfo& CurrentProcessInfo)
+void UPwLysmerAbsorbingCondition<TDim, TNumNodes>::CalculateDampingMatrix(MatrixType& rDampingMatrix, const ProcessInfo& rCurrentProcessInfo)
 {
 
     //Previous definitions
-    GeometryType& rGeom = this->GetGeometry();
+    GeometryType& r_geom = this->GetGeometry();
 
-    GeometryData::IntegrationMethod rIntegrationMethod = this->mThisIntegrationMethod;
-    const GeometryType::IntegrationPointsArrayType& IntegrationPoints = rGeom.IntegrationPoints(rIntegrationMethod);
-    const unsigned int NumGPoints = IntegrationPoints.size();
-    const unsigned int LocalDim = rGeom.LocalSpaceDimension();
+    GeometryData::IntegrationMethod r_integration_method = this->mThisIntegrationMethod;
+    const GeometryType::IntegrationPointsArrayType& r_integration_points = r_geom.IntegrationPoints(r_integration_method);
+    const unsigned int num_g_points = r_integration_points.size();
+    const unsigned int local_dim = r_geom.LocalSpaceDimension();
 
     //Containers of variables at all integration points
-    const Matrix& NContainer = rGeom.ShapeFunctionsValues(rIntegrationMethod);
-    GeometryType::JacobiansType JContainer(NumGPoints);
-    for (unsigned int i = 0; i < NumGPoints; ++i)
-        (JContainer[i]).resize(TDim, LocalDim, false);
-    rGeom.Jacobian(JContainer, rIntegrationMethod);
+    const Matrix& r_n_container = r_geom.ShapeFunctionsValues(r_integration_method);
+    GeometryType::JacobiansType j_container(num_g_points);
+    for (unsigned int i = 0; i < num_g_points; ++i)
+        (j_container[i]).resize(TDim, local_dim, false);
+    r_geom.Jacobian(j_container, r_integration_method);
 
     //Condition variables
-    BoundedMatrix<double, TDim, N_DOF> Nu = ZeroMatrix(TDim, N_DOF);
+    BoundedMatrix<double, TDim, N_DOF> nu_matrix = ZeroMatrix(TDim, N_DOF);
 
-    NormalLysmerAbsorbingVariables rVariables;
-    this->GetVariables(rVariables, CurrentProcessInfo);
+    NormalLysmerAbsorbingVariables r_variables;
```

Comment:
Also here the prefix `r_` is probably not intended (since it is not a reference).
```suggestion
    NormalLysmerAbsorbingVariables absorbing_variables;
```